api/envoy/config/core/v3/base.proto

```diff
@@ -313,6 +313,23 @@ message HeaderMap {
   repeated HeaderValue headers = 1;
 }
 
+// A file that is watched for changes, e.g. by inotify on Linux.
+message WatchedFile {
+  // Path to watch. This should either contain *filename* below or
+  // *subdirectory*/*filename* if *subdirectory* is non-empty. The watch only
+  // triggers on a move event occurring in *watch_root*. For
+  // example, if *watch_root* is */foo*, *subdirectory* is a symlink *bar* and *filename*
+  // is *baz*, ``ln -s <some_path> /foo/new && mv -Tf /foo/new /foo/bar`` will
+  // trigger a reload of ``/foo/bar/baz``.
+  string watch_root = 1 [(validate.rules).string = {min_len: 1}];
+
+  // An optional subdirectory containing the watched file.
```

Comment:
Can you clarify whether this can be multiple sub-directories with the OS specific path splitter between them? Or just a single one?

Reply:
I could go either way on this; it would be simplest to make it a single subdirectory, but if we want the versatility here of arbitrary structure under the symlink, which we might in other potential use cases, then we can make it multiple. The implementation difference is that we need to do something like `realpath(3)` instead of `readlink(2)`. On Windows I think this path canonicalization is quite complex (CC @wrowe).